model/flow/version_beacon.go

```diff
@@ -155,3 +156,43 @@ func (v *VersionBeacon) String() string {
 	}
 	return buffer.String()
 }
+
+// ProtocolStateVersionUpgrade is a service event emitted by the FlowServiceAccount
+// to signal an upgrade to the Protocol State version. `NewProtocolStateVersion`
+// must be strictly greater than the currently active Protocol State Version,
+// otherwise the service event is ignored.
+// If the node software supports `NewProtocolStateVersion`, then it uses the
+// specified Protocol State Version, beginning with the first block B where BOTH:
+//  1. The `ProtocolStateVersionUpgrade` service event has been sealed in B's ancestry
+//  2. B.view >= `ActiveView`
+//
+// NOTE: A ProtocolStateVersionUpgrade event `E` is only accepted while processing block `B`
+// which seals `E` if E.ActiveView < B.View + SafetyThreshold.
+// SafetyThreshold is a protocol parameter set so that it is overwhelmingly likely that
+// at least block is finalized within any stretch of SafetyThreshold-many blocks.
```

Comment:
I feel we should sharpen this definition further:
* We are using views here, which is the natural choice in my oppionion. Though the wording of "blocks" leads people to think about "height".

```suggestion
// SafetyThreshold is a protocol parameter set so that it is overwhelmingly likely that
// block `B` is finalized (ergo the protocol version switch at the specified view `E.ActiveView`)
// within any stretch of SafetyThreshold-many views.
```

Reply:
👍 This is accurate w.r.t. the protocol parameter as it exists. The "freeze period" parameter used by EN HCU, which is distinct but refers to the same idea, is measured in block heights rather than views. If and when we consolidate those, we'll need to revisit this.